src/app/3d/qgs3dmapcanvas.cpp

```diff
@@ -61,6 +61,9 @@ Qgs3DMapCanvas::Qgs3DMapCanvas( QWidget *parent )
 
   mEngine->window()->setCursor( Qt::OpenHandCursor );
   mEngine->window()->installEventFilter( this );
+
+  if (QgsWindow3DEngine *window = dynamic_cast< QgsWindow3DEngine * >( mEngine ))
+    window->setSize( mContainer->width(), mContainer->height() );
```

Comment:
with setSize() being virtual method, we would also avoid these unpleasant dynamic casts :-)  (which by the way are better to switch to qobject_cast for QObject-based classes)